app/Lib/Tools/BackgroundJobsTool.php

```diff
@@ -0,0 +1,681 @@
+        } else {
+            $name = $id;
+        }
+
+        $this->validateWorkerName($name);
+
+        return $this->Supervisor->stopProcess(
+            sprintf(
+                '%s:%s',
+                self::MISP_WORKERS_PROCESS_GROUP,
+                $name
+            ),
+            $waitForRestart
+        );
+    }
+
+    /**
+     * Restarts workers
+     *
+     * @param boolean $waitForRestart
+     * @return void
+     */
+    public function restartWorkers(bool $waitForRestart = false): void
+    {
+        $this->Supervisor->stopProcessGroup(self::MISP_WORKERS_PROCESS_GROUP, $waitForRestart);
+        $this->Supervisor->startProcessGroup(self::MISP_WORKERS_PROCESS_GROUP, $waitForRestart);
+    }
+
+    /**
+     * Restarts workers with status != RUNNING
+     *
+     * @param boolean $waitForRestart
+     * @return void
+     */
+    public function restartDeadWorkers(bool $waitForRestart = false): void
+    {
+        $this->Supervisor->startProcessGroup(self::MISP_WORKERS_PROCESS_GROUP, $waitForRestart);
+    }
+
+    /**
+     * Purge queue
+     *
+     * @param string $queue
+     * @return void
+     */
+    public function purgeQueue(string $queue): void
+    {
+        $this->validateQueue($queue);
+
+        $this->RedisConnection->del($queue);
+    }
+
+    /**
+     * Validate queue
+     *
+     * @return boolean
+     * @throws InvalidArgumentException
+     */
+    private function validateQueue(string $queue): bool
+    {
+        if (!in_array($queue, self::VALID_QUEUES)) {
+            throw new InvalidArgumentException(
+                sprintf(
+                    'Invalid background job queue, must be one of: [%s]',
+                    implode(', ', self::VALID_QUEUES)
+                )
+            );
+        }
+
+        return true;
+    }
+
+    /**
+     * Validate command
+     *
+     * @return boolean
+     * @throws InvalidArgumentException
+     */
+    private function validateCommand(string $command): bool
+    {
+        if (!in_array($command, self::ALLOWED_COMMANDS)) {
+            throw new InvalidArgumentException(
+                sprintf(
+                    'Invalid command, must be one of: [%s]',
+                    implode(', ', self::ALLOWED_COMMANDS)
+                )
+            );
+        }
+
+        return true;
+    }
+
+    /**
+     * Validate worker name
+     *
+     * @return boolean
+     * @throws InvalidArgumentException
+     */
+    private function validateWorkerName(string $name): bool
+    {
+        list($queue, $id) = explode('_', $name);
+
+        $this->validateQueue($queue);
+
+        if (!$this->validateQueue($queue) || !is_numeric($id)) {
+            throw new InvalidArgumentException('Invalid worker name, must be one of format {queue_name}_{process_id}, example: default_00');
+        }
+
+        return true;
+    }
+
+    /**
+     * @return Redis
+     */
+    private function createRedisConnection(): Redis
```

Comment:
You can use `$redis->setOption(Redis::OPT_SERIALIZER, Redis::SERIALIZER_JSON);`, so values will be automatically serialized/deserialized.